### PR TITLE
Sidecar example

### PR DESCRIPTION
**Pull request by CookieComputing**
Added a self-contained sidecar example demonstrating that `kube-janitor` can be run as a sidecar. I've encountered a scenario where instead of having `kube-janitor` run stand on its own, my group and I wanted to try using `kube-janitor` as a sidecar, and I figured it would be worth sharing.

`kube-janitor` will clean up any pods annotated with `janitor/ttl` in `kube-public`, including the pod hosting `kube-janitor` itself.

To test this in `kube-public`:
- Run `kubectl apply -f rbac.yaml`
- Run `kubectl create -f deployment.yaml`

### CODE REVIEW

**Comment by hjacobs** [comment]
Creating a Pod directly is not really encouraged I would say (e.g. it would be killed on cluster changes like autoscaling). Also your file is called `deployment.yaml`, so maybe change it to a real `Deployment`?

**Comment by CookieComputing** [comment]
Changed it to use a deployment, and removed the shared volume between the two containers. Let me know what you think

File: examples/sidecar/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    spec:
+      serviceAccount: kube-janitor
+      containers:
+        - name: nginx-controller
+          image: nginx
+        - name: kube-janitor-sidecar
+          image: hjacobs/kube-janitor:0.3
+          args:
+            - --interval=60
+            - --include-resources=pods,deployments,replicasets
```
Comment (hjacobs) [inline examples/sidecar/deployment.yaml]: we can restrict it to `deployments` in this case (the deployment will clean up the replicaset and pods)

Comment (CookieComputing) [reply examples/sidecar/deployment.yaml]: when I changed it to `--include-resources=deployments`, it only seems like the deployment was deleted, and not the replicaset nor the pod

**Pull request merged**